src/util/Taggable.hpp

```diff
@@ -96,6 +96,18 @@ class BaseTagDecorator {
         decorator.decorate(os);
         return os;
     }
+
+    /**
+     * @brief Gets the string representation of the tag.
+     *
+     * @return The string representation of the tag
+     */
+    std::string str() const
```

Comment:
Maybe lets give it some more verbose name, for example `toString()`?

Reply:
Sure, thanks 👍